src/validations/html/htmlInspector.js

```diff
@@ -38,8 +39,37 @@ const errorMap = {
       },
     };
   },
+  'invalid-list-children': (error) => {
+    const tag = error.context.tagName.toLowerCase();
+
+    return {
+      reason: 'invalid-list-children',
+      payload: {
+        tag,
+        children: 'li',
+      },
+    };
+  },
 };
 
+function invalidListChildrenValidator(listener, reporter) {
+  listener.on('element', (elementName, domElement) => {
+    if (domElement.childNodes.length &&
+        (elementName === 'ul' || elementName === 'ol')) {
+      // Iterate over direct descendents of lists
```

Comment:
I generally like to avoid inline comments like this. The general principle is that, if the line-by-line working of the code requires further explanation, that suggests an opportunity to improve self-documentation by giving variables, functions, and classes more descriptive names (and possibly factoring out additional functions and variables whose main purpose is to self-describe). In this case, I think your code is perfectly clear without the comments here and a few lines below.